File: Codeforces/1419D1_sages_birthday.cpp
// Solution by Inés Dávila Herrero.
// URL: https://codeforces.com/problemset/problem/1419/D1

#include <bits/stdc++.h>

using namespace std;

int main(){
    int n;
    cin >> n;

    vector<int> list(n);
    for(int i=0; i<n; i++){
        cin >> list[i];
    }
    sort(list.begin(), list.end());
    
    int r = n / 2;
    if(n % 2 == 0){
        r--;
    }

    cout << r << endl;

    for(int i=0; i<n/2; i++){
        cout << list[i + n/2] << ' ' << list[i] << ' ';
    }

    if(n % 2 == 1){
        cout << list[n - 1];
    }
}

File: Codeforces/1419D1_sages_birthday.py
"""
Solution by Inés Dávila Herrero. C++ to Python 3 by Pablo Dávila Herrero.
URL: https://codeforces.com/problemset/problem/1419/D1
"""

n = int(input())
ls = list(map(int, input().split()))

ls.sort()

r = n // 2
if n % 2 == 0:
    r -= 1

print(r)

shift = n // 2
for i in range(n // 2):
    print(ls[i + shift], ls[i], end=" ")

if n % 2 == 1:
    print(ls[-1])
